src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs

```diff
@@ -13,28 +13,31 @@ namespace Microsoft.Identity.Web
     /// </summary>
     public class ManagedIdentityClientAssertion : ClientAssertionProviderBase
     {
+        private readonly TokenCredential _credential;
+
         /// <summary>
         /// See https://aka.ms/ms-id-web/certificateless.
         /// </summary>
-        /// <param name="managedIdentityClientId"></param>
+        /// <param name="managedIdentityClientId">Optional ClientId of the Managed Identity or Workload Identity</param>
         public ManagedIdentityClientAssertion(string? managedIdentityClientId)
         {
-            _managedIdentityClientId = managedIdentityClientId;
+            _credential = new DefaultAzureCredential(
+                new DefaultAzureCredentialOptions
+                {
+                    ManagedIdentityClientId = managedIdentityClientId,
+                    WorkloadIdentityClientId = managedIdentityClientId
```

Comment:
Let's use the best of both suggestions (@psignoret and yours):
- let's use both the ManagedIdentity and WorkloadIdentity credentials
- but let's disable the credentials that don't make sense when Managed identity is expected

```suggestion
                    WorkloadIdentityClientId = managedIdentityClientId,
                    ExcludeAzureCliCredential = true,
                    ExcludeAzureDeveloperCliCredential = true
                    ExcludeAzurePowerShellCredential = true,
                    ExcludeInteractiveBrowserCredential = true,
                    ExcludeSharedTokenCacheCredential = true,
                    ExcludeVisualStudioCodeCredential = true,
                    ExcludeVisualStudioCredential = true
```